indra/llcharacter/llik.h

```diff
@@ -439,16 +454,17 @@ class Joint
     ptr_t getSingleActiveChild();
     const LLVector3& getBone() const { return mBone; }
     const LLVector3& getLocalPos() const { return mLocalPos; }
+    const LLVector3& getLocalScale() const { return mLocalScale; }
     F32 getBoneLength() const { return mBone.length(); }
     F32 getLocalPosLength() const { return mLocalPosLength; }
 
     ptr_t getParent() { return mParent; }
-    void activate() { mIsActive = true; }
-    bool isActive() const { return mIsActive; }
-    bool hasDisabledConstraint() const { return (mConfigFlags & FLAG_DISABLE_CONSTRAINT) > 0; }
+    void activate() { mIkFlags |= IK_FLAG_ACTIVE; }
```

Comment:
are these ever deactivated ?

Reply:
**Joints** are not deactivated during the IK calculations, but all activations are cleared before the IK starts its next `solve()`.  See `LLIK::Joint::resetFlags()` and where it is called.